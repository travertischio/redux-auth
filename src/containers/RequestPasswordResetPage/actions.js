/*
 *
 * RequestPasswordResetPage actions
 *
 */

import {
  REQUEST_PASSWORD_RESET_ACTION,
  REQUEST_PASSWORD_RESET_SUCCESS_ACTION,
  REQUEST_PASSWORD_RESET_FAILED_ACTION,
  DESTROY_PAGE_ACTION,
} from './constants';

export function requestPasswordResetAction(requestPasswordPayload) {
  return {
    type: REQUEST_PASSWORD_RESET_ACTION,
    payload: requestPasswordPayload,
  };
}

export function requestPasswordResetSuccessAction(response) {
  return {
    type: REQUEST_PASSWORD_RESET_SUCCESS_ACTION,
    payload: response,
  };
}

export function requestPasswordResetFailedAction(rejection) {
  return {
    type: REQUEST_PASSWORD_RESET_FAILED_ACTION,
    payload: rejection,
  };
}

export function destroyPageAction() {
  return {
    type: DESTROY_PAGE_ACTION,
  };
}
