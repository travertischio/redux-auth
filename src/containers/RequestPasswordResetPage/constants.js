/*
 *
 * RequestPasswordResetPage constants
 *
 */

export const REQUEST_PASSWORD_RESET_ACTION = 'app/RequestPasswordResetPage/REQUEST_PASSWORD_RESET';
export const REQUEST_PASSWORD_RESET_SUCCEED_ACTION = 'app/RequestPasswordResetPage/REQUEST_PASSWORD_RESET_SUCCEED';
export const REQUEST_PASSWORD_RESET_FAILED_ACTION = 'app/RequestPasswordResetPage/REQUEST_PASSWORD_RESET_FAILED';
export const DESTROY_PAGE_ACTION = 'app/RequestPasswordResetPage/DESTROY';
