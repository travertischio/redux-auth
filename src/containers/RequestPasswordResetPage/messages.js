/*
 * RequestPasswordResetPage Messages
 *
 * This contains all the text for the RequestPasswordResetPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'app.containers.RequestPasswordResetPage.pageTitle',
    defaultMessage: 'Password Change',
  },
  pageDescription: {
    id: 'app.containers.RequestPasswordResetPage.pageDescription',
    defaultMessage: 'Password Change',
  },
  header: {
    id: 'app.containers.RequestPasswordResetPage.header',
    defaultMessage: 'Let’s find your account',
  },
  successHeader: {
    id: 'app.containers.RequestPasswordResetPage.successHeader',
    defaultMessage: 'We just emailed you a link',
  },
  successMessage: {
    id: 'app.containers.RequestPasswordResetPage.successMessage',
    defaultMessage: 'Please check your email and click the secure link.',
  },
  successMessageSpamTip: {
    id: 'app.containers.RequestPasswordResetPage.successMessageSpamTip',
    defaultMessage: 'If you don’t see our email, check your spam folder. Still having trouble? Get troubleshooting tips.',
  },
  serverErrorUnknown: {
    id: 'app.containers.RequestPasswordResetPage.serverErrorUnknown',
    defaultMessage: 'An error occurred during sending request password rest. Please try again.',
  },
});
