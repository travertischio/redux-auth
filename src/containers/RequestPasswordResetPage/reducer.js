/*
 *
 * RequestPasswordResetPage reducer
 *
 */

import { fromJS } from 'immutable';
import {
  REQUEST_PASSWORD_RESET_ACTION,
  REQUEST_PASSWORD_RESET_SUCCEED_ACTION,
  REQUEST_PASSWORD_RESET_FAILED_ACTION,
  DESTROY_PAGE_ACTION,
} from './constants';

const initialState = fromJS({});

function requestPasswordResetPageReducer(state = initialState, action) {
  switch (action.type) {
    case REQUEST_PASSWORD_RESET_ACTION:
      return onRequestPasswordResetAction(state);
    case REQUEST_PASSWORD_RESET_SUCCEED_ACTION:
      return onRequestPasswordResetSucceedAction(state);
    case REQUEST_PASSWORD_RESET_FAILED_ACTION:
      return onRequestPasswordResetFailedAction(state);
    case DESTROY_PAGE_ACTION:
      return onDestroyPageAction(state);
    default:
      return state;
  }
}

function onRequestPasswordResetAction(state) {
  return state.merge({
    loading: true,
    errorMessage: false,
    sent: false,
  });
}

function onRequestPasswordResetSucceedAction(state) {
  return state.merge({
    loading: false,
    errorMessage: false,
    sent: true,
  });
}

function onRequestPasswordResetFailedAction(state) {
  return state.merge({
    loading: false,
    errorMessage: 'serverErrorUnknown',
  });
}

function onDestroyPageAction() {
  return fromJS({});
}

export default requestPasswordResetPageReducer;
