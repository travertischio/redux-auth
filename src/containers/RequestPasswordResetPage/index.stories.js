import React from 'react';
import { storiesOf } from '@kadira/storybook';
import RequestPasswordResetPage from './';

storiesOf('Request Password Reset Page', module)
  .add('Request Password Reset Page', () => <RequestPasswordResetPage />);
