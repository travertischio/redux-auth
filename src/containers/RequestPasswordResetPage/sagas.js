import {
  takeLatest,
  take,
  cancel,
  put,
  call,
} from 'redux-saga/effects';
import { LOCATION_CHANGE } from 'react-router-redux';
import { requestPasswordReset as requestPasswordResetApiCall } from '../../api';
import {
  requestPasswordResetSuccessAction,
  requestPasswordResetFailedAction,
} from './actions';
import { REQUEST_PASSWORD_RESET_ACTION } from './constants';

export function* defaultSaga() {
  const RequestPasswordResetActionWatcher = yield takeLatest(REQUEST_PASSWORD_RESET_ACTION, requestPasswordReset);

  yield take(LOCATION_CHANGE);
  yield cancel(RequestPasswordResetActionWatcher);
}

export function* requestPasswordReset(action) {
  try {
    const response = yield call(requestPasswordResetApiCall, action.payload);
    yield put(requestPasswordResetSuccessAction(response));
  } catch (error) {
    yield put(requestPasswordResetFailedAction(error));
  }
}

export default [
  defaultSaga,
];
