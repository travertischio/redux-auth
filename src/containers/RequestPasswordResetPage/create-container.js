import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { injectIntl } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import selectRequestPasswordResetPage from './selectors';
import messages from './messages';
import {
  requestPasswordResetAction,
  destroyPageAction,
  } from './actions';

export default function createRequestPasswordResetContainer(PageComponent, options = {}) {
  const mapStateToProps = createStructuredSelector({
    RequestPasswordResetPage: selectRequestPasswordResetPage,
  });

  const mapDispatchToProps = {
    onSubmitForm: requestPasswordResetAction,
    onUnMount: destroyPageAction,
  };

  @connect(mapStateToProps, mapDispatchToProps)
  @injectIntl
  class RequestPasswordResetContainer extends PureComponent {
    static propTypes = {
      intl: PropTypes.object.isRequired,
      RequestPasswordResetPage: PropTypes.object.isRequired,
      onUnMount: PropTypes.func.isRequired,
    };

    componentWillUnmount() {
      this.props.onUnMount();
    }

    getErrorMessage() {
      let errorMessage = this.props.RequestPasswordResetPage.errorMessage;

      if (errorMessage) {
        errorMessage = messages[errorMessage];
      }

      return errorMessage;
    }

    render() {
      const { formatMessage } = this.props.intl;
      const pageTitle = formatMessage(messages.pageTitle);
      const pageDescription = formatMessage(messages.pageDescription);

      return (
        <div className={options.className || 'request-password-reset-page'}>
          {!options.noHelmet &&
            <Helmet
              title={pageTitle}
              meta={[
                { name: 'description', content: pageDescription },
              ]}
            />
          }

          <PageComponent errorMessage={this.getErrorMessage()} {...this.props} />
        </div>
      );
    }
  }

  return RequestPasswordResetContainer;
}
