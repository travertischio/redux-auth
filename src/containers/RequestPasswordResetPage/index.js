/*
 *
 * RequestPasswordResetPage
 *
 */
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import createRequestPasswordResetContainer from './create-container';
import RequestPasswordResetForm from '../../components/RequestPasswordResetForm';
import messages from './messages';

class RequestPasswordResetPage extends PureComponent {
  static propTypes = {
    RequestPasswordResetPage: PropTypes.object,
    onSubmitForm: PropTypes.func,
    errorMessage: PropTypes.string,
  };

  renderSuccessMessage() {
    return (
      <div>
        <h1><FormattedMessage {...messages.successHeader} /></h1>
        <p><FormattedMessage {...messages.successMessage} /></p>
        <p><FormattedMessage {...messages.successMessageSpamTip} /></p>
      </div>
    );
  }

  renderRequestPasswordResetForm() {
    const {
      onSubmitForm,
      RequestPasswordResetPage: {
        loading,
      },
    } = this.props;

    const errorMessage = this.props.errorMessage;

    return (
      <div>
        <h1><FormattedMessage {...messages.header} /></h1>

        { loading && <div>Processing... Please wait.</div> }
        { errorMessage && <p><FormattedMessage {...errorMessage} /></p> }

        <RequestPasswordResetForm onSubmit={onSubmitForm} />
      </div>
    );
  }

  render() {
    const sent = this.props.RequestPasswordResetPage.sent;

    if (sent) {
      return this.renderSuccessMessage();
    }

    return this.renderRequestPasswordResetForm();
  }
}


export default createRequestPasswordResetContainer(RequestPasswordResetPage);
