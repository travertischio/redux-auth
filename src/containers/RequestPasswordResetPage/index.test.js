import React from 'react';
import {
  getStoreWithInitialState,
  findActionByType,
  mountWithIntl,
  createComponentWithIntl,
} from 'react-unit-testing-utils';
import RequestPasswordResetPage from './index';
import { destroyPageAction } from './actions';
import {
  REQUEST_PASSWORD_RESET_ACTION,
  DESTROY_PAGE_ACTION,
 } from './constants';

describe('<RequestPasswordResetPage />', () => {
  it('should dispatch action REQUEST_PASSWORD_RESET_ACTION when fill and submit the form', () => {
    const store = getStoreWithInitialState({
      auth: {
        isAuthenticated: false,
      },
      requestPasswordResetPage: {},
    });

    const wrapper = mountWithIntl(<RequestPasswordResetPage />, {
      store,
    });

    const email = 'team@arabel.la';

    wrapper.find('input[name="email"]').simulate('change', { target: { value: email } });
    wrapper.find('form').simulate('submit');

    const recivedAction = findActionByType(store, REQUEST_PASSWORD_RESET_ACTION);

    expect(recivedAction.type).toEqual(REQUEST_PASSWORD_RESET_ACTION);
  });

  it('should dispatch action DESTROY_PAGE_ACTION when the component unmount', () => {
    const initialState = {
      auth: {
        isAuthenticated: false,
      },
      requestPasswordResetPage: {},
    };
    const { component, store } = createComponentWithIntl(<RequestPasswordResetPage />, initialState);
    component.unmount();
    const recivedAction = findActionByType(store, DESTROY_PAGE_ACTION);
    expect(recivedAction).toEqual(destroyPageAction());
  });

  it('should NOT render RequestPasswordResetPage when user IS authenticated', () => {
    const initialState = {
      auth: {
        isAuthenticated: true,
        user: {},
      },
    };
    const { component } = createComponentWithIntl(<RequestPasswordResetPage />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render RequestPasswordResetPage when user IS NOT authenticated', () => {
    const initialState = {
      auth: {
        isAuthenticated: false,
      },
      requestPasswordResetPage: {},
    };
    const { component } = createComponentWithIntl(<RequestPasswordResetPage />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render RequestPasswordResetPage with loading indicator when loading is in the props', () => {
    const initialState = {
      auth: {
        isAuthenticated: false,
      },
      requestPasswordResetPage: {
        loading: true,
      },
    };
    const { component } = createComponentWithIntl(<RequestPasswordResetPage />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render RequestPasswordResetPage with success message when sent request', () => {
    const initialState = {
      auth: {
        isAuthenticated: false,
      },
      requestPasswordResetPage: {
        sent: true,
      },
    };
    const { component } = createComponentWithIntl(<RequestPasswordResetPage />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render RequestPasswordResetPage with error message when errorMessage is in the props', () => {
    const initialState = {
      auth: {
        isAuthenticated: false,
      },
      requestPasswordResetPage: {
        errorMessage: 'serverErrorUnknown',
      },
    };
    const { component } = createComponentWithIntl(<RequestPasswordResetPage />, initialState);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
