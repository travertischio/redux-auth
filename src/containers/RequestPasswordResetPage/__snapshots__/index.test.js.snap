// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestPasswordResetPage /> should NOT render RequestPasswordResetPage when user IS authenticated 1`] = `null`;

exports[`<RequestPasswordResetPage /> should render RequestPasswordResetPage when user IS NOT authenticated 1`] = `
<div
  className="request-password-reset-page"
>
  <div>
    <h1>
      <span>
        Let’s find your account
      </span>
    </h1>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="email"
        >
          E-mail
        </label>
        <div
          className=""
        >
          <input
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="E-mail"
            type="email"
            value=""
          />
        </div>
      </div>
      <div>
        <button
          disabled={true}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<RequestPasswordResetPage /> should render RequestPasswordResetPage with error message when errorMessage is in the props 1`] = `
<div
  className="request-password-reset-page"
>
  <div>
    <h1>
      <span>
        Let’s find your account
      </span>
    </h1>
    <p>
      <span>
        An error occurred during sending request password rest. Please try again.
      </span>
    </p>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="email"
        >
          E-mail
        </label>
        <div
          className=""
        >
          <input
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="E-mail"
            type="email"
            value=""
          />
        </div>
      </div>
      <div>
        <button
          disabled={true}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<RequestPasswordResetPage /> should render RequestPasswordResetPage with loading indicator when loading is in the props 1`] = `
<div
  className="request-password-reset-page"
>
  <div>
    <h1>
      <span>
        Let’s find your account
      </span>
    </h1>
    <div>
      Processing... Please wait.
    </div>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="email"
        >
          E-mail
        </label>
        <div
          className=""
        >
          <input
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="E-mail"
            type="email"
            value=""
          />
        </div>
      </div>
      <div>
        <button
          disabled={true}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<RequestPasswordResetPage /> should render RequestPasswordResetPage with success message when sent request 1`] = `
<div
  className="request-password-reset-page"
>
  <div>
    <h1>
      <span>
        We just emailed you a link
      </span>
    </h1>
    <p>
      <span>
        Please check your email and click the secure link.
      </span>
    </p>
    <p>
      <span>
        If you don’t see our email, check your spam folder. Still having trouble? Get troubleshooting tips.
      </span>
    </p>
  </div>
</div>
`;
